src/main/scala/supporters/BuiltinDomainsContributor.scala

```diff
@@ -110,9 +112,9 @@ abstract class BuiltinDomainsContributor extends PreambleContributor[Sort, Domai
     })
   }
 
-  protected def collectFunctions(domains: Set[ast.Domain], program: ast.Program): Unit = {
-    domains foreach (
-      _.functions foreach (df =>
+  protected def collectFunctions(domains: Set[(ast.DomainType, ast.Domain)], program: ast.Program): Unit = {
```

Comment:
This change doesn't seem to accomplish anything?

Reply:
When I was trying to add the definitions for `Map<$Snap~_$Snap>` to the SMT2 preamble I was working on that class and modified it. For this method in particular I removed the following line and instead modified this function here.

```scala
val sourceDomainInstantiations = sourceDomainInstantiationsWithType.map(x => x._2)
```

The changes in that class are purely cosmetic and not necessary for my PR. Should I revert all changes in this file?

Reply:
Ah, I see. No that seems fine then.